crypto/src/main/scala/org/bitcoins/crypto/ECDigitalSignature.scala

```diff
@@ -77,6 +77,23 @@ sealed abstract class ECDigitalSignature extends NetworkElement {
     rBytes ++ sBytes
   }
 
+  def hashType: Option[HashType] = {
+    val trailingBytes = bytes.drop(ECDigitalSignature.fromRS(r, s).bytes.length)
+
+    if (trailingBytes.nonEmpty && trailingBytes.length <= 4) {
```

Comment:
Can HashType end up being serialized to more than one byte? I thought not ... if not then this can be simplified to just taking the head of `trailingBytes`

Reply:
When serializing the hashtype for inclusion in the hash signed by the digital signature the hash if 4 bytes: https://github.com/bitcoin-s/bitcoin-s/blob/07514e2348fef55a9c2a5d4d93201c03dde7396e/core/src/main/scala/org/bitcoins/core/crypto/TransactionSignatureSerializer.scala#L117

Reply:
Looking more into it we should never expect a `HashType` beyond one byte in an `ECDigitalSignature` but this is really more of a serialization concern that is picked up by `isStrictEncoded`, so I will leave this as is. (p.s. looks like this stems from Satoshi using a `uint32_t` instead of a `uint8_t`, for no apparent reason)